File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/PassiveDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.Passive;

import java.util.List;

@Dao
public interface PassiveDao {

    @Query("SELECT * FROM passives ORDER BY id")
    LiveData<List<Passive>> getPassives();

    @Query("SELECT * FROM passives WHERE id = :passiveId")
    LiveData<Passive> getPassiveById(int passiveId);

    @Query("SELECT * FROM passives WHERE championId = :championId")
    LiveData<Passive> getPassiveByChampionId(int championId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertPassive(Passive passive);

    @Delete
    void deletePassive(Passive passive);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updatePassive(Passive passive);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/PassiveDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.Passive

@Dao
interface PassiveDao {
    @Query("SELECT * FROM passives ORDER BY id")
    fun getPassives(): LiveData<List<Passive>>

    @Query("SELECT * FROM passives WHERE id = :passiveId")
    fun getPassiveById(passiveId: Int): LiveData<Passive>

    @Query("SELECT * FROM passives WHERE championId = :championId")
    fun getPassiveByChampionId(championId: Int): LiveData<Passive>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPassive(passive: Passive)

    @Delete
    fun deletePassive(passive: Passive)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updatePassive(passive: Passive)
}